components/select/select-base.ts

```diff
@@ -70,6 +70,16 @@ export abstract class SuiSelectBase<T, U> implements AfterContentInit {
     }
 
     public set options(options:T[]) {
+        if (typeof (options) == "function") {
+            this.searchService.optionsLookup = options;
+            let subscription = this.dropdownService.isOpenChange.subscribe((value:boolean) => {
```

Comment:
I've just added some code [here](https://github.com/loudandwicked/ng2-semantic-ui/pull/1) that sorts this issue out in a more elegant way - I've explained it there.